shell/browser/electron_browser_client.cc

```diff
@@ -769,6 +770,13 @@ void ElectronBrowserClient::AppendExtraCommandLineSwitches(
     content::WebContents* web_contents =
         GetWebContentsFromProcessID(process_id);
     if (web_contents) {
+      // Extension background pages don't have WebContentsPreferences, but they
+      // support WebSQL by default.
+#if BUILDFLAG(ENABLE_ELECTRON_EXTENSIONS)
+      if (extensions::GetViewType(web_contents) !=
+          extensions::VIEW_TYPE_INVALID)
+        command_line->AppendSwitch(switches::kEnableWebSQL);
```

Comment:
Perhaps
```suggestion
      if (extensions::ProcessMap::Get(browser_context)->Contains(process->GetID()))
        command_line->AppendSwitch(switches::kEnableWebSQL);
```

By analogy with //chrome/browser/extensions/chrome_content_browser_client_extensions_part.cc (ChromeContentBrowserClientExtensionsPart::AppendExtraRendererCommandLineSwitches)

I think it's logically more correct to associate this with the _process_ rather than the WebContents, though the effect will presently be the same, I think.

Reply:
So I assume it should be out of the `if (web_contents)` block?

Reply:
yep, also you can get the BrowserContext similarly to this: https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/chrome_content_browser_client.cc;l=2189-2193;drc=398ae53c39c1011dfc62bc95e59edbdf62f66374